syncmaster/settings/database.py

```diff
@@ -0,0 +1,51 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+import textwrap
+from urllib.parse import urlparse, urlunparse
+
+from pydantic import BaseModel, Field
+
+
+class DatabaseSettings(BaseModel):
+    """Database connection settings.
+
+    .. note::
+
+        You can pass here any extra option supported by
+        `SQLAlchemy Engine class <https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine>`_,
+        even if it is not mentioned in documentation.
+
+    Examples
+    --------
+
+    .. code-block:: bash
+
+        DATABASE_URL=postgresql+asyncpg://postgres:postgres@localhost:5432/syncmaster
+        # custom option passed directly to engine factory
+        DATABASE_POOL_PRE_PING=True
+    """
+
+    url: str = Field(
+        description=textwrap.dedent(
+            """
+            Database connection URL.
+
+            See `SQLAlchemy documentation <https://docs.sqlalchemy.org/en/20/core/engines.html#backend-specific-urls>`_
+
+            .. warning:
+
+                Only async drivers are supported, e.g. ``asyncpg``
+            """,
+        ),
+    )
+
+    class Config:
+        extra = "allow"
+
+    @property
+    def sync_url(self) -> str:
```

Comment:
TODO: replace asyncpg + psycopg2 with psycopg==3.x, as it supports both async and sync

Reply:
[due to falling tests with psycopg 3.1.19](https://github.com/MobileTeleSystems/syncmaster/actions/runs/11597822360/job/32292235887) I brought back asyncpg in backend application

Reply:
Anyway, that's was just a recommendation to add `TODO` to the sources :)